### ISSUE TITLE
Algebraic_real_d_1::refine_to()  Is this a bug?

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#1594]
We get the warning below in [this testsuite](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.10-Ic-26/Algebraic_kernel_d/TestReport_lrineau_CentOS7-Release.gz):

> include/CGAL/Algebraic_kernel_d/Algebraic_real_d_1.h:324:29: 
> warning: variable 's' set but not used [-Wunused-but-set-variable]
>    CGAL::Comparison_result s;

but looking at the [code](https://github.com/CGAL/cgal/blob/master/Algebraic_kernel_d/include/CGAL/Algebraic_kernel_d/Algebraic_real_d_1.h#L312)  I am wondering if the function does what the documentation says, as it is const, I don't see anything that is mutable, and it is also not a predicate as it has void as return type.

@akobel , @ericberberich, @mhsaar : Any comment is welcome.

**Comment by ericberberich** [comment]
it eventually calls (in line 250):` this->ptr()->compare(*(y.ptr()), distinct); `, which sets the representation of the ref-counted object - which also explains that the const is fine.

**Comment by ericberberich** [comment]
You can probably rewrite it as:

`CGAL_assertion_code(CGAL::Comparison_result sl = ) compare_distinct(lo); (CGAL::LARGER  == sl);`
`CGAL_assertion_code(CGAL::Comparison_result sh = ) compare_distinct(hi); (CGAL::SMALLER == sh);`

(have not compiled it!)

**Comment by akobel** [comment]
```
CGAL_assertion_code(CGAL::Comparison_result sl = ) compare_distinct(lo); CGAL_assertion_code(CGAL::LARGER == sl;)
CGAL_assertion_code(CGAL::Comparison_result sh = ) compare_distinct(hi); CGAL_assertion_code(CGAL::SMALLER == sh;)
```
Otherwise, I agree with Eric. Perhaps you should fake-use the output of `compare_distinct` when assertions are unused to prevent a warning. (I haven't compiled it, either.)

Explanation: The function will change the internal representation of the algebraic number (refine the boundaries of the interval that represents the number), but does not change the value and/or semantics of the number.

**Comment by afabri** [comment]
Thank you for your explanations. I added the assertions.

**Comment by lrineau** [comment]
I reopen the issue. It will be closed as usual once the PR #1596 will be *merge* into `master`.
